src/WCSimWCAddDarkNoise.cc

```diff
@@ -0,0 +1,254 @@
+#include "WCSimWCAddDarkNoise.hh"
+#include "WCSimWCPMT.hh"
+#include "WCSimWCDigi.hh"
+#include "WCSimWCHit.hh"
+#include "G4EventManager.hh"
+#include "G4Event.hh"
+#include "G4SDManager.hh"
+#include "G4DigiManager.hh"
+#include "G4ios.hh"
+#include "G4RotationMatrix.hh"
+#include "G4ThreeVector.hh"
+#include "G4LogicalVolumeStore.hh"
+
+#include "WCSimDetectorConstruction.hh"
+#include "WCSimPmtInfo.hh"
+#include "WCSimDarkRateMessenger.hh"
+
+#include <vector>
+#include <utility>
+// for memset
+#include <cstring>
+#include <iostream>
+
+#ifndef WCSIMWCADDDARKNOISE_VERBOSE
+//#define WCSIMWCADDDARKNOISE_VERBOSE
+#endif
+
+WCSimWCAddDarkNoise::WCSimWCAddDarkNoise(G4String name,
+                                   WCSimDetectorConstruction* myDetector)
+  :G4VDigitizerModule(name)
+{
+  this->myDetector = myDetector;
+  DigiHitMap.clear();
```

Comment:
Ok, let's try again post-caffeine. My points were:
- Despite my earlier comment, now that I see that you are setting the Boolean in SetParameterName to true, I am surprised that the value is not being set to the default if not specified in the macro. I thought that's what that Boolean did. 
- Since the Boolean is subtle (and I guess not working like I thought it did), I am on board with explicitly setting it in the constructor like you are doing. You should probably set that Boolean in SetParameterName to false if you are going to do that, though.